### PR TITLE
[RNMobile] Refactor blockWidth setting and passing down

### PR DESCRIPTION
**Pull request by lukewalczak**
## Description

That PR is extending the [merged part](https://github.com/WordPress/gutenberg/commit/1e90244af08459a56c84c8d083ba159f52ce9039) and fixes and improves the logic for setting and passind down by props the block width.

Ref to gb-mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3052

 

## How has this been tested?

* Open a draft post
* Add `Cover` block and choose the color

**Expect the paragraph block is visible and editable**

* Add `Buttons` block
* Type some characters in the first button and add the next one

**Expect two buttons are visible and editable**

* Add `Group` block
* Nest other block within

**Expect nested block is displayed properly**

* Add `SocialLinks` block

**Expect block is displayed properly**

* Add `Media&Text` block

**Expect block is displayed properly**

* Add `Columns` block
* Choose 30/70 variation
* Nest something in both columns
* Set full width alignment in `Columns`

**Expect block along with nested blocks is displayed properly**

* Rotate the device

**Expect blocks extends properly**

## Types of changes

Bug fix - fixes the issue with undefined block width within inner blocks.

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation. 
- [X] I've included developer documentation if appropriate. 
- [X] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by geriux — APPROVED** [review]
LGTM! Tested it on an iPhone, iPad (with split-view), and Android.

Nice work Luke!

File: packages/block-editor/src/components/block-list/block.native.js [hunk 1]
```diff
@@ -121,29 +123,28 @@ class BlockListBlock extends Component {
...
 
 	getBlockForType() {
+		const { blockWidth } = this.state;
 		return (
 			<BlockForType
 				{ ...this.props }
 				onBlockFocus={ this.onFocus }
 				insertBlocksAfter={ this.insertBlocksAfter }
 				getBlockWidth={ this.getBlockWidth }
+				blockWidth={ blockWidth }
```
Comment (dratwas) [inline packages/block-editor/src/components/block-list/block.native.js:147]: I think that if we use the `blockWidth` only here there is no need for destructuring it :)

File: packages/block-editor/src/components/block-list/index.native.js [hunk 2]
```diff
@@ -372,6 +377,7 @@ export default compose( [
 				blockClientIds = filterInnerBlocks( blockClientIds );
 			}
 
+			const { maxWidth } = getSettings();
 			const isReadOnly = getSettings().readOnly;
```
Comment (dratwas) [inline packages/block-editor/src/components/block-list/index.native.js:381]:
```suggestion
			const { maxWidth, readOnly: isReadOnly } = getSettings();
```

**Review by dratwas — APPROVED** [review]
Everything seems fine ;) I have one small idea, WDYT about passing the `blockWidth` through context? It's a primitive variable so there will not be an issue with a new references etc.

**Comment by cameronvoell** [comment]
Tests all green after CI restart. I see it's approved so I'll go ahead and merge.

**Pull request merged**